src/lsdb/core/crossmatch/abstract_crossmatch_algorithm.py

```diff
@@ -58,3 +61,27 @@ def crossmatch(self) -> pd.DataFrame:
     def _rename_columns_with_suffix(dataframe, suffix):
         columns_renamed = {name: name + suffix for name in dataframe.columns}
         dataframe.rename(columns=columns_renamed, inplace=True)
+
+    @classmethod
+    def _append_extra_columns(
+        cls, dataframe: pd.DataFrame, extra_columns: Dict[str, pd.Series] | None = None
+    ):
+        """Adds crossmatch extra columns to the resulting Dataframe."""
+        if cls.extra_columns is None:
+            return
+        if extra_columns is None:
+            raise ValueError("No extra column values were provided")
```

Comment:
@hombit can confirm here, but I think it's possible to not have any extra columns, e.g. in the case of SFD matching.  In that case, we're just adding in full row from the SFD catalog, and there's no meaningful distance or match likelihood.

Is the intent then that some `SFDCrossmatch` just never calls `_append_extra_columns`?

Reply:
Yes, that's exactly what I was thinking!